### PR TITLE
Remove DevTools dependency on Scheduler runWithPriority

### PR DESCRIPTION
**Pull request by bvaughn**
### Do not land before PR #20955

Should not have any observable user impact.

### CODE REVIEW

**Review by rickhanlonii — APPROVED** [review]
Awesome thanks!

**Comment by bvaughn** [comment]
Yup! If you think about it just ping me once you land your other PR

**Comment by rickhanlonii** [comment]
@bvaughn the other PR has landed, but CI looks stuck 🤔

File: packages/react-devtools-shared/src/devtools/views/Components/TreeContext.js [hunk 1]
```diff
@@ -923,11 +919,10 @@ function TreeContextController({
...
     (action: Action) => {
-      // Run the first update at "user-blocking" priority in case dispatch is called from a non-React event.
-      // In this case, the current (and "next") priorities would both be "normal",
-      // and suspense would potentially block both updates.
-      runWithPriority(UserBlockingPriority, () => dispatch(action));
-      next(() => dispatch({type: 'UPDATE_INSPECTED_ELEMENT_ID'}));
+      dispatch(action);
+      startTransition(() => {
+        dispatch({type: 'UPDATE_INSPECTED_ELEMENT_ID'});
+      });
```
Comment (gaearon) [inline packages/react-devtools-shared/src/devtools/views/Components/TreeContext.js:925]: 😍

Comment (bvaughn) [reply packages/react-devtools-shared/src/devtools/views/Components/TreeContext.js:925]: Yeah. This is really much nicer. :)

**Pull request merged**